make install fails with "Version must be a string"

I got a error in `make install` .
```
[bnf-mode (master)]$ make install
Loading 00debian-vars...
Loading /etc/emacs/site-start.d/50autoconf.el (source)...
Loading /etc/emacs/site-start.d/50cmake-data.el (source)...
Loading /etc/emacs/site-start.d/50coq.el (source)...
Loading /etc/emacs/site-start.d/50dictionaries-common.el (source)...
Loading debian-ispell...
Loading /var/cache/dictionaries-common/emacsen-ispell-default.el (source)...
Loading /var/cache/dictionaries-common/emacsen-ispell-dicts.el (source)...
Parsing tar file...
Parsing tar file...done
Version must be a string
Makefile:125: recipe for target 'install' failed
make: *** [install] Error 255
```
I'm using Emacs 25.2.2 on Ubuntu 18.04 LTS.

--- a/.elpaignore
+++ b/.elpaignore
@@ -1,4 +1,4 @@
-.travis.yml
+.github
 .ert-runner
 .gitignore
 Makefile

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -87,7 +87,7 @@ jobs:
       - name: Run Unit Tests
         run: make test
 
-      - name: Build Reporting
+      - name: After Success Reporting
         if: success()
         run: git log --format=fuller -5
 

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -62,7 +62,7 @@ jobs:
           sudo rm -f /etc/apt/sources.list.d/azure*.list
 
           sudo apt-get update --quiet --yes 1>/dev/null
-          sudo apt-get install texinfo bsdtar pandoc
+          sudo apt-get install texinfo pandoc
 
       - name: Initialize the Project
         run: make init
@@ -72,7 +72,8 @@ jobs:
 
       - name: Minimal Package Test
         run: |
-          test "$(tar --list --file bnf-mode-${VERSION}.tar | wc -l)" = 7
+          make install
+          head "$HOME/.emacs.d/elpa/bnf-mode-$VERSION/bnf-mode.el" | grep -q "$VERSION"
 
       - name: Create Release
         id: create_release

--- a/Cask
+++ b/Cask
@@ -7,8 +7,7 @@
 
 (files "*.el")
 
-(depends-on "rx")
-(depends-on "cl-lib")
+(depends-on "cl-lib" "0.5")
 
 (development
  (depends-on "f")

--- a/Makefile
+++ b/Makefile
@@ -114,7 +114,7 @@ test:
 .PHONY: clean
 clean:
 	$(CASK) clean-elc
-	$(RM) -f README ChangeLog $(ARCHIVE_NAME).info
+	$(RM) -f README ChangeLog $(ARCHIVE_NAME).info coverage-final.json
 	$(RM) -f $(ARCHIVE_NAME)-pkg.el $(ARCHIVE_NAME)-*.tar
 
 .PHONY: package

--- a/Makefile
+++ b/Makefile
@@ -23,7 +23,7 @@ ROOT_DIR := $(shell dirname $(realpath $(lastword $(MAKEFILE_LIST))))
 EMACS ?= emacs
 CASK ?= cask
 PANDOC ?= pandoc
-TAR ?= COPYFILE_DISABLE=1 bsdtar
+TAR ?= tar
 
 EMACSFLAGS ?=
 TESTFLAGS ?= --reporter ert+duration
@@ -89,7 +89,7 @@ $(ARCHIVE_NAME)-pkg.el: $(ARCHIVE_NAME).el
 	$(CASK) pkg-file
 
 $(PACKAGE_NAME).tar: README ChangeLog LICENSE $(ARCHIVE_NAME).el $(ARCHIVE_NAME)-pkg.el $(ARCHIVE_NAME).info dir
-	$(TAR) -c -s "@^@$(PACKAGE_NAME)/@" -f $(PACKAGE_NAME).tar $^
+	$(TAR) -c -v -f $(PACKAGE_NAME).tar --transform "s@^@$(PACKAGE_NAME)/@" $^
 
 # Public targets
 
@@ -122,7 +122,8 @@ package: $(PACKAGE_NAME).tar
 
 .PHONY: install
 install: $(PACKAGE_NAME).tar
-	$(EMACS) --batch -l package -f package-initialize --eval "(package-install-file \"$(PWD)/$(PACKAGE_NAME).tar\")"
+	$(EMACS) --batch -l package -f package-initialize --eval \
+		"(let ((debug-on-error t))(package-install-file \"$(PWD)/$(PACKAGE_NAME).tar\"))"
 
 .PHONY: help
 help: .title

--- a/NEWS
+++ b/NEWS
@@ -6,6 +6,12 @@ See the end of the file for license conditions.
 This file is about changes in BNF Mode.
 
 * BNF Mode 0.4.3
+** Fixed dependency list in the Cask file.
+Dependency list was fixed by specifying `cl-lib' version as well as
+removing `rx` (`rx' is part of GNU Emacs actually).  This fixed the
+issue reported in <https://github.com/sergeyklay/bnf-mode/issues/5>.
+The issue related to installation from the TAR file.
+
 ** CI/CD process was moved on GitHub Actions.
 
 * BNF Mode 0.4.2

--- a/README.org
+++ b/README.org
@@ -70,7 +70,7 @@ in ~bnf-mode~.
 To customize various options, use command as follows:
 
 #+begin_src
-M-x customize-group bnf RET
+M-x customize-group RET bnf RET
 #+end_src
 
 ** Support

--- a/dir
+++ b/dir
@@ -15,4 +15,4 @@ File: dir,	Node: Top	This is the top of the INFO tree
 * Menu:
 
 Languages
-* BNF Mode: (bnf-mode).               Major mode for editing BNF grammars.
+* BNF Mode: (bnf-mode).         Major mode for editing BNF grammars.

